### PR TITLE
add extmod_blacklist to make it easier to remove modules

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Similar to extmod_whitelist, but blacklist instead

### What issues does this PR fix or reference?
https://github.com/saltstack/zh/issues/1059

### Tests written?

Yes

### CODE REVIEW

File: salt/runners/saltutil.py [hunk 1]
```diff
@@ -69,10 +72,11 @@ def sync_modules(saltenv='base', extmod_whitelist=None):
 
         salt-run saltutil.sync_modules
     '''
-    return salt.utils.extmods.sync(__opts__, 'modules', saltenv=saltenv, extmod_whitelist=extmod_whitelist)[0]
+    return salt.utils.extmods.sync(__opts__, 'modules', saltenv=saltenv, extmod_whitelist=extmod_whitelist
```
Comment (cachedout) [inline salt/runners/saltutil.py]: This has a syntax error. Please add a trailing comma.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
The test written for this appears to have failed: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/8458/testReport/junit/integration.modules.saltutil/SaltUtilSyncModuleTest/test_sync_all_blacklist/

**Comment by rallytime** [comment]
@gtmanfred Looks like the following related tests are failing here. Can you take another look?

- `integration.modules.saltutil.SaltUtilSyncModuleTest.test_sync_all_blacklist_and_whitelist `
- `integration.modules.saltutil.SaltUtilSyncModuleTest.test_sync_all_blacklist`

**Comment by gtmanfred** [comment]
Whoops! Forgot to add the blacklist to the sync_all function, thanks tests!

**Comment by gtmanfred** [comment]
hrm, now they are all failing

**Comment by gtmanfred** [comment]
Ok, they should all pass on this run.

Thanks!

**Comment by cachedout** [comment]
Afraid not. :] Please have another look at the tests. Thanks, @gtmanfred

**Comment by gtmanfred** [comment]
I rebased my changes on develop, to see if that fixes those extra errors.  I am seeing everything passing when I run the full test suite.  We will see what jenkins says.

**Pull request merged**